File: VB/ChartsDemo.Wpf/ViewModels/G7Data.vb
Imports System
Imports System.Collections.Generic
Imports System.Globalization
Imports System.IO
Imports System.Linq
Imports System.Windows
Imports System.Windows.Media
Imports System.Xml
Imports System.Xml.Serialization

Namespace ChartsDemo
    <XmlRoot("G7GDPs")>
    Public Class G7Data
        Inherits List(Of GDP)


        Private Shared members_Renamed As List(Of G7Member)

        Private Shared data_Renamed As List(Of GDP)

        Public Shared ReadOnly Property Members() As List(Of G7Member)
            Get
                If members_Renamed IsNot Nothing Then
                    Return members_Renamed
                Else
                    members_Renamed = GetG7Data()
                    Return members_Renamed
                End If
            End Get
        End Property
        Public Shared ReadOnly Property Data() As List(Of GDP)
            Get
                If data_Renamed IsNot Nothing Then
                    Return data_Renamed
                Else
                    data_Renamed = GetGDPData()
                    Return data_Renamed
                End If
            End Get
        End Property

        Private Shared Function GetG7Data() As List(Of G7Member)
            Return Data.GroupBy(Function(x) x.Country).Select(Function(x) New G7Member(x.Key, x.ToList())).ToList()
        End Function
        Private Shared Function GetGDPData() As List(Of GDP)
            Dim s = New XmlSerializer(GetType(G7Data))
            Return DirectCast(s.Deserialize(DataLoader.LoadFromResources("/Data/GDPofG7.xml")), List(Of GDP))
        End Function
    End Class

    Public Class G7Member
        Public ReadOnly Property GDPin2015() As Decimal
            Get
                Return GDPs.First(Function(gdp) gdp.Year = 2015).Product
            End Get
        End Property
        Public Property CountryName() As String
        Private privateGDPs As List(Of GDP)
        Public Property GDPs() As List(Of GDP)
            Get
                Return privateGDPs
            End Get
            Private Set(ByVal value As List(Of GDP))
                privateGDPs = value
            End Set
        End Property

        Public Sub New(ByVal countryName As String, ByVal items As List(Of GDP))
            Me.CountryName = countryName
            Me.GDPs = items
        End Sub
    End Class

    Public Class GDP
        Public Property Country() As String
        Public Property Year() As Integer
        Public Property Product() As Decimal
    End Class
End Namespace

File: CS/ChartsDemo.Wpf/ViewModels/G7Data.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Windows;
using System.Windows.Media;
using System.Xml;
using System.Xml.Serialization;

namespace ChartsDemo {
    [XmlRoot("G7GDPs")]
    public class G7Data : List<GDP> {
        static List<G7Member> members;
        static List<GDP> data;

        public static List<G7Member> Members { get { return members ?? (members = GetG7Data()); } }
        public static List<GDP> Data { get { return data ?? (data = GetGDPData()); } }

        static List<G7Member> GetG7Data() {
            return Data.GroupBy(x => x.Country).Select(x => new G7Member(x.Key, x.ToList())).ToList();
        }
        static List<GDP> GetGDPData() {
            var s = new XmlSerializer(typeof(G7Data));
            return (List<GDP>)s.Deserialize(DataLoader.LoadFromResources("/Data/GDPofG7.xml"));
        }
    }

    public class G7Member {
        public decimal GDPin2015 { get { return GDPs.First((gdp) => gdp.Year == 2015).Product; } }
        public string CountryName { get; set; }
        public List<GDP> GDPs { get; private set; }

        public G7Member(string countryName, List<GDP> items) {
            this.CountryName = countryName;
            this.GDPs = items;
        }
    }

    public class GDP {
        public string Country { get; set; }
        public int Year { get; set; }
        public decimal Product { get; set; }
    }
}
